DocsCode/NavigationTutorial/NavigationTutorialViewController.m

```diff
@@ -0,0 +1,150 @@
+@import MapboxNavigation;
+@import MapboxDirections;
+// #-end-code-snippet: navigation dependencies-objc
+
+@interface NavigationTutorialViewController () <MGLMapViewDelegate>
+
+@property (nonatomic) MBNavigationMapView *mapView;
+// #-code-snippet: navigation directions-route-objc
+@property (nonatomic) MBRoute *directionsRoute;
+// #-end-code-snippet: navigation directions-route-objc
+
+
+@end
+
+@implementation NavigationTutorialViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    // #-code-snippet: navigation init-map-objc
+    self.mapView = [[MBNavigationMapView alloc] initWithFrame:self.view.bounds];
+
+    [self.mapView setCenterCoordinate:CLLocationCoordinate2DMake(30.265, -97.741)
+                            zoomLevel:11 animated:NO];
+    [self.view addSubview:self.mapView];
+    // Set the map view's delegate
+    self.mapView.delegate = self;
+    // #-end-code-snippet: navigation init-map-objc
+    
+    // #-code-snippet: navigation user-location-objc
+    // Allow the map view to display the user's location
+    self.mapView.showsUserLocation = YES;
+    // #-end-code-snippet: navigation user-location-objc
+    
+    // #-code-snippet: navigation gesture-recognizer-objc
+    // Add a gesture recognizer to the map view
+    UITapGestureRecognizer *setDestination = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(didLongPress:)];
+    [self.mapView addGestureRecognizer:setDestination];
+    // #-end-code-snippet: navigation gesture-recognizer-objc
+}
+
+// #-code-snippet: navigation allow-callouts-objc
+// Implement the delegate method that allows annotations to show callouts when tapped
+-(BOOL)mapView:(MGLMapView *)mapView annotationCanShowCallout:(id<MGLAnnotation>)annotation {
+    return true;
+}
+// #-end-code-snippet: navigation allow-callouts-objc
+
+// #-code-snippet: navigation long-press-objc
+-(void)didLongPress:(UITapGestureRecognizer *)sender {
+    if (sender.state != UIGestureRecognizerStateEnded) {
+        return;
+    }
+    
+    // Converts point where user did a long press to map coordinates
+    CGPoint point = [sender locationInView:self.mapView];
+    CLLocationCoordinate2D coordinate = [self.mapView convertPoint:point toCoordinateFromView:self.mapView];
+    
+    // Create a basic point annotation and add it to the map
+    MGLPointAnnotation *annotation = [MGLPointAnnotation alloc];
+    annotation.coordinate = coordinate;
+    annotation.title = @"Start navigtation";
+    [self.mapView addAnnotation:annotation];
+    
+    // Calcuate the route from the user's location to the set destination
+    [self calculateRoutefromOrigin:self.mapView.userLocation.coordinate
+                     toDestination:annotation.coordinate
+                        completion:^(MBRoute * _Nullable route, NSError * _Nullable error) {
+                            if (error != nil) {
+                                printf("Error calculating route");
```

Comment:
Use `NSLog()` in ObjC.

Reply:
Also, since we have the `error` object, let’s print that, too. ([Like this.](https://github.com/mapbox/MapboxDirections.swift/blob/11313a6ce9909117992a6655f79a4e739ed93f30/Directions%20Example/ViewController.m#L41).